### ISSUE TITLE
Empty project opened after pressing enter to confirm project name change

### ISSUE THREAD

**Issue by xvcgreg** [enso-org/enso#7589]
### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

When I finished typing new name for the project I have pressed Enter to confirm the name change. This started opening the project (with old name) but it opened a blank project (just like a new project with starting node). When I closed it the project had it's name renamed and when I have opened it it had all the nodes I have created previously.

https://github.com/enso-org/enso/assets/108271897/d425f8cd-b705-48e8-96f0-9af5638528d1

^^ repro recording

This is error message I have gotten after I managed to confirm the name change with the checkbox after pressing enter (that opened the project):
![image](https://github.com/enso-org/enso/assets/108271897/bc9d771b-0fad-47fc-9979-8141481be8af)
^^ error from 1st occurrence 

Another issue that happened after this weird behavior is that Enso in assets view didn't recognize the name changed and crashed on launch of this project (but I couldn't repro that).

### Expected behaviour

Enter confirming name change, nor opening the project.

### Enso Version

2023.8.15 nightly

### Browser or standalone distribution

Standalone distribution (local project)

### Browser Version or standalone distribution

standalone

### Operating System

Windows

### Operating System Version

Win11pro 22H2 22621.1848

### Hardware you are using

12th Gen Intel(R) Core(TM) i9-12900HK / RTX3060 Laptop / Nvidia Drivers 531.68

**Comment by xvcgreg** [comment]
Another issue connected to this:
When I want to change the project name back to the original one I get this error: 
![image](https://github.com/enso-org/enso/assets/108271897/cefcd1ee-ca4b-4c6b-aa1f-f6d17902a516)

Apparently it exists in the projects folder:
![image](https://github.com/enso-org/enso/assets/108271897/344aac7b-090f-4084-a305-a82719b1a856)

But it's not listed in assets.

**Comment by somebody1234** [comment]
just encountered this error.
PM json-rpc messages (formatted using diff syntax highlighting to distinguish frontend -> PM messages (green) and PM -> frontend messages (red)):

```diff
+ {"jsonrpc":"2.0","id":0,"method":"project/list","params":{}}
- {"jsonrpc":"2.0","id":0,"result":{"projects":[{"name":"New_Project_11","namespace":"local","id":"1839e9f4-6661-49da-9c1a-a10581396995","engineVersion":"2023.2.1-nightly.2023.8.16","created":"2023-08-16T06:27:47.648324Z","lastOpened":"2023-08-17T04:08:41.419095Z"},{"name":"New_Project_12","namespace":"local","id":"e88f22ab-e075-4879-b0a3-23a4377bf443","engineVersion":"2023.2.1-nightly.2023.8.16","created":"2023-08-16T06:42:50.901191Z","lastOpened":"2023-08-16T06:42:52.354855Z"},{"name":"New_Project_10","namespace":"local","id":"1f3d32cd-77fa-4a11-a5fa-64b901a1acfc","engineVersion":"2023.2.1-nightly.2023.8.16","created":"2023-08-16T06:12:24.068586Z","lastOpened":"2023-08-16T06:25:22.210322Z"},{"name":"Orders_1","namespace":"local","id":"6b6cda56-2439-4ac9-a72f-913cbe8559d6","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:19.88202Z","lastOpened":"2023-08-11T12:29:11.02064Z"},{"name":"New_Project_2","namespace":"local","id":"50602b0e-b7e9-4f4c-ba1a-2addbd4e20da","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:16.738509Z","lastOpened":"2023-08-10T18:50:40.56367Z"},{"name":"New_Project_7","namespace":"local","id":"b99d65c5-88a0-4f58-a916-57846171519e","engineVersion":"2023.2.1-nightly.2023.8.6","created":"2023-08-08T01:12:30.311814Z","lastOpened":"2023-08-10T16:48:25.251337Z"},{"name":"New_Project_7","namespace":"local","id":"c3a4879b-25f3-4732-b0a0-40503cd00e73","engineVersion":"2023.2.1-nightly.2023.8.6","created":"2023-08-08T01:12:30.311814Z","lastOpened":"2023-08-10T16:48:25.251156Z"},{"name":"New_Project_7","namespace":"local","id":"6c7faf28-3402-4952-b232-0b55461fd078","engineVersion":"2023.2.1-nightly.2023.8.6","created":"2023-08-08T01:12:30.311814Z","lastOpened":"2023-08-10T16:48:25.250958Z"},{"name":"New_Project_7","namespace":"local","id":"7e7f831a-e97b-4a0a-b269-4961ff02f366","engineVersion":"2023.2.1-nightly.2023.8.6","created":"2023-08-08T01:12:30.311814Z","lastOpened":"2023-08-10T16:48:25.250944Z"},{"name":"New_Project_7","namespace":"local","id":"79019c27-1fdc-4b79-99b2-14002256e82b","engineVersion":"2023.2.1-nightly.2023.8.6","created":"2023-08-08T01:12:30.311814Z","lastOpened":"2023-08-10T16:48:25.250605Z"},{"name":"New_Project_7","namespace":"local","id":"91d2bac9-ff0b-4b1e-8343-0d9171efe302","engineVersion":"2023.2.1-nightly.2023.8.6","created":"2023-08-10T16:48:25.242178Z","lastOpened":null},{"name":"New_Project_4","namespace":"local","id":"586694ad-5a11-45ec-8944-946b705448c4","engineVersion":"2023.2.1-nightly.2023.8.10","created":"2023-08-10T14:04:33.668559Z","lastOpened":"2023-08-10T14:04:35.768456Z"},{"name":"New_Project_3","namespace":"local","id":"cd0cda3f-b521-48c9-ab47-f983008a9a16","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-10T03:45:48.649177Z","lastOpened":"2023-08-10T04:51:30.798616Z"},{"name":"Restaurants_1","namespace":"local","id":"0f361dc8-f30e-43d3-a07e-4b297a739978","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:18.993883Z","lastOpened":"2023-08-09T16:00:07.612161Z"},{"name":"Colorado_COVID_2","namespace":"local","id":"0387d4f9-ea6a-4c8e-a521-014977290b5d","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:32.169399Z","lastOpened":"2023-08-09T14:26:33.1854Z"},{"name":"Stargazers_1","namespace":"local","id":"951fed00-00e4-498c-a371-c20a79d858db","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:18.518693Z","lastOpened":"2023-08-09T14:26:20.975712Z"},{"name":"NASDAQReturns_1","namespace":"local","id":"388fbfa7-9ed2-47ba-a3d4-d525311241f8","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:17.818779Z","lastOpened":"2023-08-09T14:26:19.884008Z"},{"name":"Colorado_COVID_1","namespace":"local","id":"eef97dcd-3440-4d10-ac14-7413fb20b0f7","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:17.459823Z","lastOpened":"2023-08-09T14:26:19.242344Z"},{"name":"KMeans_1","namespace":"local","id":"cf8a676b-f309-473b-82d2-14144488af46","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:17.069094Z","lastOpened":"2023-08-09T14:26:18.680741Z"},{"name":"New_Project_1","namespace":"local","id":"caf3e62a-a963-4636-b04f-fa4bc1990d0c","engineVersion":"2023.2.1-nightly.2023.8.9","created":"2023-08-09T14:26:16.600774Z","lastOpened":"2023-08-09T14:26:17.47597Z"}]}}
+ {"jsonrpc":"2.0","id":1,"method":"project/create","params":{"missingComponentAction":"Install","name":"New_Project_13"}}
- {"jsonrpc":"2.0","id":1,"result":{"projectId":"03966a03-3667-4e01-a85d-2faf30954c3a","projectName":"New_Project_13"}}
+ {"jsonrpc":"2.0","id":2,"method":"project/open","params":{"projectId":"03966a03-3667-4e01-a85d-2faf30954c3a","missingComponentAction":"Install"}}
+ {"jsonrpc":"2.0","id":3,"method":"project/open","params":{"projectId":"03966a03-3667-4e01-a85d-2faf30954c3a","missingComponentAction":"Install"}}
- {"jsonrpc":"2.0","id":2,"result":{"engineVersion":"2023.2.1-nightly.2023.8.16","languageServerJsonAddress":{"host":"127.0.0.1","port":60027},"languageServerBinaryAddress":{"host":"127.0.0.1","port":58484},"projectName":"New_Project_13","projectNormalizedName":"New_Project_13","projectNamespace":"local"}}
- {"jsonrpc":"2.0","id":3,"result":{"engineVersion":"2023.2.1-nightly.2023.8.16","languageServerJsonAddress":{"host":"127.0.0.1","port":60027},"languageServerBinaryAddress":{"host":"127.0.0.1","port":58484},"projectName":"New_Project_13","projectNormalizedName":"New_Project_13","projectNamespace":"local"}}
+ {"jsonrpc":"2.0","id":4,"method":"project/close","params":{"projectId":"03966a03-3667-4e01-a85d-2faf30954c3a"}}
+ {"jsonrpc":"2.0","id":5,"method":"project/rename","params":{"projectId":"03966a03-3667-4e01-a85d-2faf30954c3a","name":"Bc_08_16"}}
- {"jsonrpc":"2.0","id":5,"error":{"code":4010,"message":"Failure during renaming [code: 6001 message: Session not initialised]."}}
+ {"jsonrpc":"2.0","id":6,"method":"project/rename","params":{"projectId":"03966a03-3667-4e01-a85d-2faf30954c3a","name":"Bc_08_16"}}
- {"jsonrpc":"2.0","id":6,"error":{"code":4003,"message":"Project with the provided name exists"}}
- {"jsonrpc":"2.0","id":4,"error":{"code":1,"message":"Service error"}}
```

terminal logs:
```
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:34.148Z] [org.enso.projectmanager.protocol.ClientController] Client connected to Project Manager [34de0225-f97b-44cc-857a-4c07e78b5812]
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:35.232Z] [org.enso.projectmanager.infrastructure.log.Slf4jLogging] Created project with actual name [New_Project_13].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:35.239Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] The engine [2023.2.1-nightly.2023.8.16] found.
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:35.522Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Found GraalVM runtime [GraalVM 23.0.0-java17.0.7].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.12Z] [org.enso.projectmanager.infrastructure.log.Slf4jLogging] Project created [Project(03966a03-3667-4e01-a85d-2faf30954c3a,New_Project_13,New_Project_13,local,UserProject,2023-08-17T07:11:35.239285Z,None,/home/me/enso/projects/New_Project_13_2,None,None)].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.166Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] The engine [2023.2.1-nightly.2023.8.16] found.
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.357Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Found GraalVM runtime [GraalVM 23.0.0-java17.0.7].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.358Z] [org.enso.projectmanager.infrastructure.languageserver.LanguageServerBootLoader] Booting a language server [LanguageServerDescriptor(language-server-03966a03-3667-4e01-a85d-2faf30954c3a,2288a8d5-c6bb-4eff-a0f1-b36115fdfe01,/home/me/enso/projects/New_Project_13_2,NetworkConfig(127.0.0.1,49152,65535),org.enso.projectmanager.versionmanagement.DefaultDistributionConfiguration$@187ee435,2023.2.1-nightly.2023.8.16,JVMSettings(None,List()),false,None,None,None,Future(Success(Some(ws://localhost:35703/))),false)].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.358Z] [org.enso.projectmanager.infrastructure.languageserver.LanguageServerBootLoader] Found sockets for the language server [json:127.0.0.1:60027, binary:127.0.0.1:58484].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.359Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] The engine [2023.2.1-nightly.2023.8.16] found.
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.523Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Found GraalVM runtime [GraalVM 23.0.0-java17.0.7].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:36.853Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] The engine [2023.2.1-nightly.2023.8.16] found.
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:37.115Z] [org.enso.runtimeversionmanager.components.RuntimeVersionManager] Found GraalVM runtime [GraalVM 23.0.0-java17.0.7].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:37.289Z] [org.enso.languageserver.boot.MainModule] Initializing main module of the Language Server from [2023.2.1-nightly.2023.8.16, LanguageServerConfig(127.0.0.1,60027,58484,2288a8d5-c6bb-4eff-a0f1-b36115fdfe01,/home/me/enso/projects/New_Project_13_2,ProfilingConfig(None,None,None),StartupConfig(false),language-server,scala.concurrent.impl.ExecutionContextImpl$$anon$3@3da30852[Running, parallelism = 16, size = 1, active = 1, running = 1, steals = 0, tasks = 0, submissions = 0]), info]
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:37.274Z] [org.enso.languageserver.boot.LanguageServerComponent] Starting Language Server...
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:37.989Z] [org.enso.languageserver.runtime.RuntimeConnector] Starting the runtime connector.
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:37.989Z] [org.enso.languageserver.runtime.RuntimeConnector] Runtime connector established connection with the message endpoint [com.oracle.truffle.api.instrumentation.TruffleInstrument$Env$MessageTransportProxy$MessageEndpointProxy@6a4ec4cc].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:38.135Z] [org.enso.languageserver.search.SuggestionsHandler] Starting suggestions handler from [Config(projectContentRoot=ContentRootWithFile(contentRoot=Project(2288a8d5-c6bb-4eff-a0f1-b36115fdfe01),file=***/New_Project_13_2), fileManager=FileManagerConfig(3 seconds,16), vcsManager=VcsManagerConfig(5 seconds,3 seconds,true), pathWatcher=PathWatcherConfig(5 seconds,5 seconds,10), executionContext=ExecutionContextConfig(5 seconds), directories=DirectoriesConfig(***/New_Project_13_2)), org.enso.searcher.sql.SqlSuggestionsRepo@5286c398].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:38.198Z] [org.enso.languageserver.boot.MainModule] Main module of the Language Server initialized with config [Config(projectContentRoot=ContentRootWithFile(contentRoot=Project(2288a8d5-c6bb-4eff-a0f1-b36115fdfe01),file=***/New_Project_13_2), fileManager=FileManagerConfig(3 seconds,16), vcsManager=VcsManagerConfig(5 seconds,3 seconds,true), pathWatcher=PathWatcherConfig(5 seconds,5 seconds,10), executionContext=ExecutionContextConfig(5 seconds), directories=DirectoriesConfig(***/New_Project_13_2))].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:38.35Z] [org.enso.languageserver.boot.LanguageServerComponent] Started server at json:127.0.0.1:60027, binary:127.0.0.1:58484
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:38.652Z] [org.enso.projectmanager.infrastructure.languageserver.LanguageServerBootLoader] Language server booted [LanguageServerConnectionInfo(127.0.0.1,60027,58484)].
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:11:46.003Z] [org.enso.projectmanager.infrastructure.languageserver.ProjectRenameAction] Requesting a Language Server to rename project [New_Project_13].
 INFO ide_ci::program::command: npm⚠️ [error] [2023-08-17T07:11:46.015Z] [org.enso.projectmanager.infrastructure.languageserver.ProjectRenameAction] Error occurred during renaming project [code: 6001, message: Session not initialised]
 INFO ide_ci::program::command: npm⚠️ [error] [2023-08-17T07:11:46.016Z] [org.enso.projectmanager.requesthandler.ProjectRenameHandler] Request Number(5) failed due to LanguageServerFailure(Failure during renaming [code: 6001 message: Session not initialised].).
 INFO ide_ci::program::command: npm⚠️ [error] [2023-08-17T07:11:53.964Z] [org.enso.projectmanager.requesthandler.ProjectRenameHandler] Request Number(6) failed due to ProjectExists.
 INFO ide_ci::program::command: npm⚠️ [error] [2023-08-17T07:11:59.673Z] [org.enso.projectmanager.requesthandler.ProjectCloseHandler] Request ProjectClose with Number(4) timed out.
 INFO ide_ci::program::command: npm⚠️ [info] [2023-08-17T07:12:06.686Z] [akka.actor.LocalActorRef] Message [scala.util.Left] from Actor[akka://project-manager/user/manager-client-controller-34de0225-f97b-44cc-857a-4c07e78b5812/request-handler-ProjectClose-c56d6c32-8141-494b-b461-78d8947ffbc5#95225506] to Actor[akka://project-manager/user/manager-client-controller-34de0225-f97b-44cc-857a-4c07e78b5812/request-handler-ProjectClose-c56d6c32-8141-494b-b461-78d8947ffbc5#95225506] was not delivered. [1] dead letters encountered, no more dead letters will be logged in next [5.000 min]. If this is not an expected behavior then Actor[akka://project-manager/user/manager-client-controller-34de0225-f97b-44cc-857a-4c07e78b5812/request-handler-ProjectClose-c56d6c32-8141-494b-b461-78d8947ffbc5#95225506] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'.
```

**Comment by hubertp** [comment]
From the exchanged messages the error appears correct - GUI is trying to rename the project once the project has been closed. Hence the session initialization problem. @4e6 how does this sound to you?

**Comment by somebody1234** [comment]
iirc renaming used to work even when projects were not open, not 100% sure though.

**Comment by xvcgreg** [comment]
The project gets renamed only after closing it - it shows renamed name in the list view thou.

https://github.com/enso-org/enso/assets/108271897/e5923826-c238-4407-ab66-488b2d543b5d

**Comment by 4e6** [comment]
I reproduced the original issue. It is the issue with the welcome screen (how it handles the `enter` key).

What happens in the original issue after renaming the project and pressing the `enter` key:
- welcome screen sends `project/open` to the original project
- welcome screen sends `project/rename` to the original project
- welcome screen sends `project/create` and opens IDE with a new (empty) project while java process with the original project still runs in the background.
- when you stop the project, it stops both running projects (the original one that runs in the background and the freshly created one). After the original project is stopped, the project manager renames the directory

**Comment by 4e6** [comment]
Regarding the issue that IDE does not update the project name of the running project (in the top left corner). 

It happens because IDE is unaware of the change. The `refactoring/renameProject` request is initiated by the project manager, and the project manager gets the response with the new name (not the IDE). I will create an issue to implement a notification about the project rename action that will be sent to IDE.

**Comment by 4e6** [comment]
The issue for updating the project name in IDE is #7604

Initially, the project was renamed in IDE (by editing the name in the top left corner), and this way it was aware of the new name. After the renaming action was delegated to the welcome screen, IDE became unaware of the new name.

**Comment by somebody1234** [comment]
tried to repro but `ide watch` isn't working properly - it says `tailwind.css` not found. @Frizi may this perhaps be related to your changes to the npm workspace?

**Comment by somebody1234** [comment]
not sure how this flew under the radar... should be fixed soon though

**Comment by somebody1234** [comment]
temporarily reopening until we can confirm that this issue is indeed solved

**Comment by farmaazon** [comment]
@sylwiabr could you confirm it is fixed?

**Comment by farmaazon** [comment]
@sylwiabr bump ^

**Comment by sylwiabr** [comment]
Looks fine now
